bank_accounts.py

```diff
@@ -64,23 +82,27 @@ def create_bank_account():
 
         # Required params
         account_id = str(uuid4())
-        account_number = str(uuid4().int)[:6]
+        account_number = str(int(uuid4()))[:6]
+        logging.debug(f"Account number: {account_number}")
+        hashed_account_number = hashlib.sha256(account_number.encode()).hexdigest()
```

Comment:
Say i generate random "salt" for each account number, then i'd like to later use it inside the transfer function, to determine which account number i'm about to transfer money to. So what I need is to store that salt for each account. Will you add it as an attribute inside the db? ( meaning each account will have a column in the db that stores its unique "Salt" for later use). 

Another approach that comes to mind is to generate it one time, then store it as an environmental variable. So each account number will be salted using the same salt(less secure I know). 

Which approach will you choose ?

Reply:
The one salt, because the whole idea is that if your DB is leaked, it would be useless for the attacker. If you store the salt in the DB, on each user item, then they have the salt and can still "guess" the value using brute-force.